### PR TITLE
HCM fixes for ComboBox and Statuslight

### LINKED ISSUE #4082: ComboBox: wrong border and background colors for dropdown button with forced-colors: active
# 🐛 Bug Report 

## 😯 Current Behavior

Desktop:
ComboBox with forced-colors: active

Mobile (Note border color for text input is green, meaning disabled):
MobileComboBox with forced-colors: active

Express: 
ComboBox with Express theme with forced-colors: active

Express on Mobile  (Note border color for text input is green, meaning disabled): 
MobileComboBox with Express theme with forced-colors: active

## 🔦 Context

Accessibility Audit

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| react-spectrum   | [@adobe/react-spectrum@3.24.1](https://github.com/adobe/react-spectrum/releases/tag/%40adobe%2Freact-spectrum%403.24.1)
| Browser          | Chrome/Edge
| Operating System | Windows

## 🧢 Your Company/Team

Adobe/Accessibility

## 🕷 Tracking Issue (optional)

A11Y-5821

### LINKED ISSUE #4096: StausLight: In light theme while using HCM, text is invisible
# 🐛 Bug Report 

StatusLight component doesn't seem to use systems colors when in High Contrast Mode. See screenshots below. Text is almost invisible in HCM for the component in light, lightest themes. 

## 🤔 Expected Behavior

In HCM, across themes, text must use 'CanvasText' color so that it is visible against the background. Below is a screenshot of how it'd look if 'CanvasText' is used.

StatusLight using CanvasText in HCM

## 😯 Current Behavior

Below is a screenshot of how it currently looks.

StatusLight currently in HCM

## 💻 Example
Issue can be observed on [Storybook](https://reactspectrum.blob.core.windows.net/reactspectrum/41e26fa0f01ee9ab6e5c21edf6f7bb5d498bf40f/storybook/index.html?path=/story/statuslight--default) or [react-spectrum docs](https://react-spectrum.adobe.com/react-spectrum/StatusLight.html) or [chromatic](https://www.chromatic.com/component?appId=5f0dd5ad2b5fc10022a2e320&csfId=statuslight--all-variants&buildNumber=454&k=63f50211c92ed3657d97b744-1200-interactive-true&h=16&b=-7) while using Windows high contrast mode.

## 🌍 Your Environment

If react-spectrum team needs access to https://assistivlabs.com/ to test WHCM, please let Adobe Accessibility team know. We can get you access.

| Software         | Version(s) |
| ---------------- | ---------- |
| react-spectrum   | 3 |
| Browser          | Edge 109 |
| Operating System | Windows |

### PR DESCRIPTION
**Pull request by LFDanLu**
Closes #4082, closes #4096, closes #4114, closes https://github.com/orgs/adobe/projects/19/views/4?pane=issue&itemId=28260452

Also has some other additional fixes from things I noticed from testing (express button backgrounds, disabled background color etc)

## ✅ Pull Request Checklist:

- [ ] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

## 📝 Test Instructions:

Compare the combobox WHCM for express and non-express with the XD file found here: https://spectrum.adobe.com/page/combo-box/#Windows-high-contrast-mode

## 🧢 Your Project:

RSP

### CODE REVIEW

File: packages/@adobe/spectrum-css-temp/components/inputgroup/skin.css [hunk 1]
```diff
@@ -264,19 +265,30 @@ governing permissions and limitations under the License.
...
+      --spectrum-textfield-background-color-disabled: Canvas;
     }
     .spectrum-FieldButton {
       forced-color-adjust: auto;
+      --spectrum-fieldbutton-background-color: Canvas;
+      --spectrum-fieldbutton-quiet-background-color: Canvas;
+      --spectrum-fieldbutton-background-color-disabled: Canvas;
+      --spectrum-fieldbutton-quiet-background-color-disabled: Canvas;
+      --spectrum-fieldbutton-background-color-hover: Canvas;
+      --spectrum-fieldbutton-quiet-background-color-hover: Canvas;
```
Comment (LFDanLu) [inline packages/@adobe/spectrum-css-temp/components/inputgroup/skin.css:291]: I'm unsure about this change here for express. It stops the combobox button for non-express from changing colors on hover, unsure what the style for the express field button should be for the combobox in general

**Review by devongovett — APPROVED** [review]
lgtm, tested in Edge

**Review by reidbarber — DISMISSED** [review]
LGTM

**Review by reidbarber — APPROVED** [review]
Should the Combobox's overlay have a border color in express HCM? Otherwise looks good.

**Comment by LFDanLu** [comment]
@reidbarber ah good catch, https://github.com/adobe/react-spectrum/pull/4050 should cover it. I'll put that PR into our testing session monday

**Review by devongovett — APPROVED** [review]

**Pull request merged**